src/bin/pgcopydb/dump_restore.c

```diff
@@ -31,23 +31,58 @@ static bool copydb_copy_database_properties_hook(void *ctx,
 
 
 /*
- * copydb_objectid_has_been_processed_already returns true when a doneFile
- * could be found on-disk for the given target object OID.
+ * copydb_objectid_has_been_processed_already returns true when the given
+ * target object OID is found in our SQLite summary catalogs. This only applies
+ * to indexes or constraints, as TABLE DATA is not part of either the
+ * --pre-data parts of the schema nor the --post-data parts of the schema.
  */
 bool
-copydb_objectid_has_been_processed_already(CopyDataSpec *specs, uint32_t oid)
+copydb_objectid_has_been_processed_already(CopyDataSpec *specs,
+										   ArchiveContentItem *item)
 {
 	DatabaseCatalog *sourceDB = &(specs->catalogs.source);
 
-	bool done = false;
+	uint32_t oid = item->objectOid;
 
-	if (!summary_lookup_oid(sourceDB, oid, &done))
+	switch (item->desc)
 	{
-		/* errors have aleady been logged */
-		return false;
+		case ARCHIVE_TAG_INDEX:
+		{
+			SourceIndex index = { .indexOid = oid };
+			CopyIndexSpec indexSpecs = { .sourceIndex = &index };
+
+			if (!summary_lookup_index(sourceDB, &indexSpecs))
+			{
+				/* errors have aleady been logged */
+				return false;
+			}
+
+			return indexSpecs.summary.doneTime > 0;
```

Comment:
I like the fact that we changed our logic to decide whether or not an item is processed. It is definitely an improvement over whether or not the lookup exists in summary.